### PR TITLE
etcd-tester: do not restart stresser on error

### PR DESCRIPTION
**Pull request by xiang90**
@gyuho this simple change wont work?

### CODE REVIEW

**Comment by gyuho** [comment]
Here https://github.com/coreos/etcd/blob/master/tools/functional-tester/etcd-tester/tester.go#L163-L180

It is possible that `err == nil` in `revs, hashes, err = tt.cluster.getRevisionHash()` 


https://github.com/coreos/etcd/blob/master/tools/functional-tester/etcd-tester/tester.go#L184-L188

It is possible that `err == nil && failed == true`. In that case, we still restart stresser first, and then call `cleanup`?

**Comment by xiang90** [comment]
@gyuho That is another bug we need to fix. I think I mentioned that to you a few days ago. we should return error on that case. also the printing has issues.

**Comment by gyuho** [comment]
@xiang90 Got it.

This change LGTM. Thanks!

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -193,10 +195,10 @@ func (tt *tester) checkConsistency() (failed bool, err error) {
 func (tt *tester) compact(rev int64, timeout time.Duration) (err error) {
 	tt.cancelStressers()
 	defer func() {
-		serr := tt.startStressers()
-		if err == nil {
-			err = serr
+		if err != nil {
+			return err
 		}
+		return tt.startStressers()
```
Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: `err = tt.startStressers()`? Nvm, if it compiles.

**Comment by xiang90** [comment]
@gyuho Can you remove failed from return val (only use error). Also can you please fix the print line issue?

**Comment by gyuho** [comment]
@xiang90 Ok sounds good. Will do after this PR. Thanks.

**Pull request merged**